swarm/src/lib.rs

```diff
@@ -1277,12 +1345,13 @@ where
         transport: transport::Boxed<(PeerId, StreamMuxerBox)>,
         behaviour: TBehaviour,
         local_peer_id: PeerId,
+        executor: Option<Box<dyn Executor + Send>>,
```

Comment:
I think we should deprecate this function too in favor of two new ones:

- `with_executor`
- `without_executor`